be/src/olap/tablet.cpp

```diff
@@ -283,35 +164,70 @@ OLAPStatus Tablet::add_rowset(RowsetSharedPtr rowset, bool need_persist) {
             LOG(FATAL) << "failed to save rowset to local meta store" << rowset->rowset_id();
         }
     }
+    RETURN_NOT_OK(save_meta());
     ++_newly_created_rowset_num;
     return OLAP_SUCCESS;
 }
 
-OLAPStatus Tablet::modify_rowsets(const vector<RowsetSharedPtr>& to_add,
-                                  const vector<RowsetSharedPtr>& to_delete) {
+OLAPStatus Tablet::add_rowsets(const vector<RowsetSharedPtr>& to_add) {
+    WriteLock wrlock(&_meta_lock);
```

Comment:
should check whether the rowset with same rowset_id or with same version already exists.